Source/SwiftLintBuiltInRules/Rules/Style/RedundantDiscardableLetRule.swift

```diff
@@ -100,4 +147,56 @@ private extension AccessorBlockSyntax {
         }
         return false
     }
+
+    var isPreviewProviderBody: Bool {
+        guard let binding = parent?.as(PatternBindingSyntax.self),
+              binding.pattern.as(IdentifierPatternSyntax.self)?.identifier.text == "previews",
+              let bindingList = binding.parent?.as(PatternBindingListSyntax.self),
+              let variableDecl = bindingList.parent?.as(VariableDeclSyntax.self) else {
+            return false
+        }
+
+        guard variableDecl.modifiers.contains(keyword: .static) &&
+                variableDecl.bindingSpecifier.tokenKind == .keyword(.var) else {
+            return false
+        }
+
+        if let type = binding.typeAnnotation?.type.as(SomeOrAnyTypeSyntax.self) {
+            return type.someOrAnySpecifier.text == "some" &&
+                type.constraint.as(IdentifierTypeSyntax.self)?.name.text == "View"
+        }
+
+        return false
+    }
+}
+
+private extension CodeBlockSyntax {
+    var isViewBuilderFunctionBody: Bool {
+        parent?.as(FunctionDeclSyntax.self)?.isViewBuilderFunction == true
+    }
+}
+
+private extension FunctionDeclSyntax {
+    var isViewBuilderFunction: Bool {
+        guard attributes.contains(attributeNamed: "ViewBuilder") else {
+            return false
+        }
+
+        guard let returnType = signature.returnClause?.type.as(SomeOrAnyTypeSyntax.self) else {
+            return false
+        }
+
+        return returnType.someOrAnySpecifier.text == "some" &&
+            returnType.constraint.as(IdentifierTypeSyntax.self)?.name.text == "View"
```

Comment:
This could be a computed property on `SomeOrAnyTypeSyntax` to avoid the duplication.

Reply:
Fixed here: https://github.com/realm/SwiftLint/pull/6075/commits/5e364e7cf1446aad3f38e47acad47989bab42762